scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -321,7 +321,12 @@ class ScioContext private[scio] (val options: PipelineOptions,
         Future.successful(result.getState)
     }
 
-    new ScioResult(result, finalState, _accumulators.values.toSeq, pipeline)
+    val scioResult = new ScioResult(result, finalState, _accumulators.values.toSeq, pipeline)
+    val metricsLocation = optionsAs[ScioOptions].getMetricsLocation
+    if (metricsLocation != null && finalState.isCompleted) {
+      scioResult.saveMetrics(metricsLocation)
```

Comment:
This only works with blocking runner? Should support non-blocking ones by setting a Future handler?